Build failure: php / php82 while building php-soap (darwin)

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix build nixpkgs#php` or `nix build nixpkgs#php82` on macOS (Arm64 in my case)

### Build log

```
error: builder for '/nix/store/kpwnv9l1qyk77y4p7pbhcb0mqwxcr0m5-php-soap-8.2.23.drv' failed with exit code 2;
       last 10 log lines:
       > =====================================================================
       > FAILED TEST SUMMARY
       > ---------------------------------------------------------------------
       > SOAP Bug #71610 - Type Confusion Vulnerability - SOAP / make_http_soap_request() [tests/bug71610.phpt]
       > Bug #55639 (Digest authentication dont work) [tests/bugs/bug55639.phpt]
       > Bug #76232 (SoapClient Cookie Header Semicolon) [tests/bugs/bug76232.phpt]
       > Bug #76348 (WSDL_CACHE_MEMORY causes Segmentation fault) [tests/bugs/bug76348.phpt]
       > SOAP customized Content-Type, eg. SwA use case [tests/custom_content_type.phpt]
       > =====================================================================
       > make: *** [Makefile:136: test] Error 1
       For full logs, run 'nix log /nix/store/kpwnv9l1qyk77y4p7pbhcb0mqwxcr0m5-php-soap-8.2.23.drv'.
error: 1 dependencies of derivation '/nix/store/mf1k5wcfn7168p12iabq0lfsiw4yk84m-php-extra-init-8.2.23.ini.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hhfckjr07yx7cwi7zy73rr6lxq0l47nq-php-with-extensions-8.2.23.drv' failed to build
```

### Additional context

Building PHP 8.1, 8.3 and 8.4 works. Building PHP 8.2 on NixOS (Arm64) also works. Using the previous commit with PHP 8.2.22 works (cdce031214793d766942068e3f8e5c8c65e4b094). Not sure what causes the test failure. Also tested without sandbox.

### Notify maintainers

Was last updated by @Ma27.

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.5.0, macOS 14.5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/9qq0zf30wi74pz66rr05zmxq0nv17q1p-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/top-level/php-packages.nix
+++ b/pkgs/top-level/php-packages.nix
@@ -705,6 +705,12 @@ in {
               url = "https://github.com/php/php-src/commit/4fe821311cafb18ca8bdf20b9d796c48a13ba552.diff?full_index=1";
               hash = "sha256-YC3I0BQi3o3+VmRu/UqpqPpaSC+ekPqzbORTHftbPvY=";
             })
+          ] ++ lib.optionals (lib.versions.majorMinor php.version == "8.2") [
+            # Fix test 'bug55639.phpt'
+            (fetchpatch {
+              url = "https://github.com/php/php-src/commit/1b52ecd78ad1a211a4a9db65975df34d2539125b.patch";
+              hash = "sha256-LVk9sfwl5D+rHzyYjfV4pAuhBjSPXj1WjTfnrzBJXhY";
+            })
           ] ++ lib.optionals (lib.versions.majorMinor php.version == "8.3" && lib.versionOlder php.version "8.3.10") [
             (fetchpatch {
               url = "https://github.com/php/php-src/commit/ecf0bb0fd12132d853969c5e9a212e5f627f2da2.diff?full_index=1";

